### DISCUSSION TITLE
General purpose buzzer control

### DISCUSSION THREAD

**Discussion by ramanraja** [arendst/Tasmota#11714] [Q&A] [answered]
Does Tasmota support a buzzer only on Sonoff iFan3 ?  If so, it is a pity !
The Buzzer command seems to be so versatile and useful, I want to use it in my DIY hardware. 
I have connected a buzzer to GPIO 14 and now I want all Buzzer functionality (on-off cycle and bit mask pattern) on this pin.
Please suggest some starting point.

**Comment by ascillato** [comment, marked as answer]
It is already implemented and enabled since v6.6.0.18 (now Tasmota is on 9.3.1.2 at this moment)

It is a general buzzer not limited to iFAN03

Buzzer usage: https://tasmota.github.io/docs/Buzzer/

Command explanation: https://tasmota.github.io/docs/Commands/#control

> I have connected a buzzer to GPIO 14

So, in module option, define GPIO14 as BUZZER

**Reply by ramanraja** [reply]
Thank you. I am using 9.3.1.1 build. My hardware is derived from Base module Generic (18). I do not see the BUZZER option in the drop down list on the Configure Template page on the Web UI. 
Then I referred to the page https://tasmota.github.io/docs/Components/ for component codes and found the buzzer there.
I edited the template by hand and entered 512 for GPIO14.  (My buzzer has inverted logic, to it is Buzzeri). But it did not work. Also, the template constant explicitly mentions  "Sonoff iFan03 Buzzer inverted", so I was unsure if it will work for modules other than the iFan.

**Reply by ascillato** [reply]
Something is wrong in your custom self-compile build. Please, try again but with the precompiled version of Tasmota ( http://ota.tasmota.com/tasmota/tasmota.bin.gz )

Buzzer is included in the standard build: **_tasmota.bin_**

**Reply by ascillato** [reply]
![image](https://user-images.githubusercontent.com/35405447/114599354-5d8db800-9c69-11eb-8ac0-7b512e29045a.png)

**Reply by ramanraja** [reply]
That was a useful hint. 
#define USE_BUZZER 
#define BUZZER_ENABLE  true
in the header solved the problem. Thanks a lot.
